hbase/stackable/patches/2.6.1/0005-Allow-overriding-ipc-bind-port-and-use-alternative-p.patch

```diff
@@ -0,0 +1,195 @@
+From f2360ad266a17eb342eb826ae9dad08d3293b580 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Natalie=20Klestrup=20R=C3=B6ijezon?= <nat@nullable.se>
+Date: Fri, 30 May 2025 14:26:26 +0200
+Subject: Allow overriding ipc bind port and use alternative port from listener
+
+---
+ .../apache/hadoop/hbase/master/HMaster.java   | 11 ++++++--
+ .../hbase/regionserver/HRegionServer.java     | 28 +++++++++++++------
+ .../hbase/regionserver/RSRpcServices.java     |  4 +--
+ 3 files changed, 30 insertions(+), 13 deletions(-)
+
+diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
+index 3fe5abac27..6d40db77ae 100644
+--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
++++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
+@@ -20,6 +20,7 @@ package org.apache.hadoop.hbase.master;
+ import static org.apache.hadoop.hbase.HConstants.DEFAULT_HBASE_SPLIT_COORDINATED_BY_ZK;
+ import static org.apache.hadoop.hbase.HConstants.HBASE_MASTER_LOGCLEANER_PLUGINS;
+ import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_COORDINATED_BY_ZK;
++import static org.apache.hadoop.hbase.HConstants.MASTER_PORT;
+ import static org.apache.hadoop.hbase.master.cleaner.HFileCleaner.CUSTOM_POOL_SIZE;
+ import static org.apache.hadoop.hbase.util.DNS.MASTER_HOSTNAME_KEY;
+ 
+@@ -559,6 +560,11 @@ public class HMaster extends HRegionServer implements MasterServices {
+     return conf.get(MASTER_HOSTNAME_KEY);
+   }
+ 
++  @Override
++  protected int getUseThisPortInstead(Configuration conf) {
++    return conf.getInt(MASTER_PORT, this.rpcServices.getSocketAddress().getPort());
```

Comment:
One caveat here is that previously you could set `hbase.master.port=0` to automatically allocate the port number, and the advertisement would automatically be updated to the correct port.

This patch still _allows_ that behaviour, but requires you to instead set `hbase.master.ipc.port=0` and _unset_ `hbase.master.port`. A workaround would to use something like:

```suggestion
+    int port = conf.getInt(MASTER_PORT, 0);
+    return port != 0 ? port : this.rpcServices.getSocketAddress().getPort();
```